archaius2-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java

```diff
@@ -15,299 +15,261 @@
- *  Multibinder.newSetBinder(binder(), ConfigResolver.class, RuntimeLayer.class)...
+ *  {@literal @}RuntimeLayer SettableConfig config
  *  
- * Override properties may be set by either injecting 
- *  @OverrideLayer CompositeConfig config
- * or at startup by specifying a multibinding
- *  Multibinder.newSetBinder(binder(), ConfigResolver.class, OverrideLayer.class)...
+ * A remote configuration may be specified by binding to {@literal @}RemoteLayer Config
+ * When setting up a remote configuration that need access to archaius's Config
+ * make sure to inject the qualifier {@literal @}Raw otherwise the injector will fail
+ * with a circular dependency error.  Note that the injected config will have 
+ * system, environment and application properties loaded into it.
  * 
- * Application properties will be loaded automatically from resources with the base 
- * name indicated by the binding to @ApplicationLayer String.  The default value
- * is 'application' be can be changed via a binding override.
- * 
- * @author elandau
+ * <code>
+ * public class FooRemoteModule extends AbstractModule {
+ *     {@literal @}Override
+ *     protected void configure() {}
+ *     
+ *     {@literal @}Provides
+ *     {@literal @}RemoteLayer
+ *     Config getRemoteConfig({@literal @}Raw Config config) {
+ *         return new FooRemoteConfigImplementaiton(config);
+ *     }
+ * }
+ * </code>
  */
 public final class ArchaiusModule extends AbstractModule {
+    private static final String DEFAULT_CONFIG_NAME     = "application";
     
-    private static final String RUNTIME_LAYER_NAME              = "RUNTIME";
-    private static final String REMOTE_LAYER_NAME               = "REMOTE";
-    private static final String SYSTEM_LAYER_NAME               = "SYSTEM";
-    private static final String ENVIRONMENT_LAYER_NAME          = "ENVIRONMENT";
-    private static final String APPLICATION_OVERRIDE_LAYER_NAME = "APPLICATION_OVERRIDE";
-    private static final String APPLICATION_LAYER_NAME          = "APPLICATION";
-    private static final String LIBRARIES_LAYER_NAME            = "LIBRARIES";
-    private static final String DEFAULTS_LAYER_NAME             = "DEFAULTS";
+    private static final String RUNTIME_LAYER_NAME      = "RUNTIME";
+    private static final String REMOTE_LAYER_NAME       = "REMOTE";
+    private static final String SYSTEM_LAYER_NAME       = "SYSTEM";
+    private static final String ENVIRONMENT_LAYER_NAME  = "ENVIRONMENT";
+    private static final String APPLICATION_LAYER_NAME  = "APPLICATION";
+    private static final String LIBRARIES_LAYER_NAME    = "LIBRARIES";
     
-    private static final AtomicInteger idCounter = new AtomicInteger();
+    static {
+        System.setProperty("archaius.default.configuration.class",      "com.netflix.archaius.bridge.StaticAbstractConfiguration");
+        System.setProperty("archaius.default.deploymentContext.class",  "com.netflix.archaius.bridge.StaticDeploymentContext");
```

Comment:
Yes, but there were some static initialization ordering issues with code that was previously invoked as part of archaius2 bootstrapping which triggered the wrong classes being created in archaius1.